metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage.py

```diff
@@ -297,20 +448,23 @@ def _populate_view_downstream_lineage(self, conn: SnowflakeConnection) -> None:
                 view_name: str = self.get_dataset_identifier_from_qualified_name(
                     db_row["VIEW_NAME"]
                 )
+                downstream_table: str = self.get_dataset_identifier_from_qualified_name(
+                    db_row["DOWNSTREAM_TABLE_NAME"]
+                )
                 if not self._is_dataset_pattern_allowed(
                     view_name, db_row["VIEW_DOMAIN"]
+                ) or not self._is_dataset_pattern_allowed(
+                    downstream_table, db_row["DOWNSTREAM_TABLE_DOMAIN"]
                 ):
                     continue
-                downstream_table: str = self.get_dataset_identifier_from_qualified_name(
-                    db_row["DOWNSTREAM_TABLE_NAME"]
-                )
+
                 # Capture view->downstream table lineage.
-                self._lineage_map[downstream_table].append(
+                self._lineage_map[downstream_table].update_lineage(
                     # (<upstream_view_name>, <json_list_of_upstream_view_columns>, <json_list_of_downstream_columns>)
-                    (
+                    SnowflakeUpstreamTable.from_dict(
                         view_name,
-                        db_row["VIEW_COLUMNS"],
-                        db_row["DOWNSTREAM_TABLE_COLUMNS"],
+                        json.loads(db_row["VIEW_COLUMNS"]),
```

Comment:
what happens if this bombs?

Reply:
hmm, I don't expect bombing since the `VIEW_COLUMNS` column is equivalent to `columns` json array as per snowflake [access history view](https://docs.snowflake.com/en/sql-reference/account-usage/access_history.html) documentation and the dict to class conversion is protected already.